Automattic/vip-cli
# SQL Import: Poll for import status after kick off & in separate command

## Description

Uses the job status from GraphQL to inform the user about the status of the import as it works through the system.

It leverages both the jobs system and the import_status site meta to relay the progress of the current import and to report failures to the user.

## Steps to Test

1. Check out PR.
1. Run `npm run build`
1. Run an import e.g. `./dist/bin/vip-import-sql.js @1000.production --search-replace="from.example.com,to.example.com" ~/Downloads/dbdump.sql`
    * Confirm job status polls appropriately
    * Confirm a successful import is reported correctly
1. Repeat the above, and exit the CLI prior to job completion (or proceed in a separate terminal session)
1. Run `./dist/bin/vip-import-sql-status.js @1000.production`
    * Confirm job status polls appropriately as above
1. Attempt to import various files that you know will fail server-side. (e.g. manually alter a db dump file to insert a row with a duplicate unique key)
    * Confirm the error is provided to you and is actionable.
